src/components/CancelSwitchPopup/CancelSwitchPopup.tsx

```diff
@@ -22,29 +23,33 @@ const CancelSwitchPopup = () => {
   const [isError, setIsError] = useState(false);
   const [step, setStep] = useState(1);
 
-  const { data: allSwitchDetails } = useSelector(
+  const { data: allSwitchDetails } = useAppSelector(
     (state) => state.plan.switchDetails
   );
+
+  const { cancelSwitch } = useAppSelector((state) => state.popupManager);
+
   const {
-    cancelSwitch: {
-      pendingSwitchId,
-      switchDirection,
-      switchOfferTitle: untranslatedSwitchOfferTitle,
-      baseOfferTitle: untranslatedBaseOfferTitle,
-      baseOfferExpirationDate,
-      baseOfferPrice
-    }
-  } = useSelector((state) => state.popupManager);
+    pendingSwitchId,
+    switchDirection,
+    switchOfferTitle: untranslatedSwitchOfferTitle,
+    baseOfferTitle: untranslatedBaseOfferTitle,
+    baseOfferExpirationDate,
+    baseOfferPrice
+  } = cancelSwitch as CancelSwitch;
```

Comment:
isnt `cancelSwitch` typed? casting a type here seems like a band-aid. please take a look at the store state slice and type that correctly, so all further usages will contain proper types automatically

Reply:
I casted it because it it's type in store is `CancelSwitch | null` but yea, it's not the best idea. I will return `MyAccountError` component if `cancelSwitch` is null to get rid of casting it